File: 백준/Bronze/5086. 배수와 약수/배수와 약수.py
import sys
input = sys.stdin.readline

ans = ['factor', 'multiple', 'neither']

while True:
    A, B = map(int, input().split())

    if A == 0 and B == 0:
        break

    if B % A == 0:
        print(ans[0])
    elif A % B == 0:
        print(ans[1])
    else:
        print(ans[2])

File: 백준/Bronze/5086. 배수와 약수/배수와 약수.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int n = sc.nextInt();
            int m = sc.nextInt();

            if ((n == 0) && (m == 0)) {
                break;
            }

            String ans = "";
            if (m % n == 0) {
                ans = "factor";
            } else if (n % m == 0) {
                ans = "multiple";
            } else {
                ans = "neither";
            }

            System.out.println(ans);
        }
    }
}
